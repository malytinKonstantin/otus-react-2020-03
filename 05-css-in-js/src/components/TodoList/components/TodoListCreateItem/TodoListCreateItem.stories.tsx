import React from 'react'
import { action } from '@storybook/addon-actions'
import { withKnobs, array } from '@storybook/addon-knobs'
import { TodoListCreateItem } from './TodoListCreateItem'

export default {
  title: 'Todo list create item',
  decorators: [withKnobs],
}

export const TodoListCreateItemExample: React.FC<{}> = () => {
  return <TodoListCreateItem onCreate={action('create item')} />
}
