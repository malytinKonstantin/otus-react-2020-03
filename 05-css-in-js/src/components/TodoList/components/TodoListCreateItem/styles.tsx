import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 15px;

  & * {
    margin-right: 15px;
  }
`
export const TextField = styled.input`
  padding: 5px 10px;
  border-radius: 5px;
  box-shadow: none;
  border: 1px solid #afafaf;
`
export const Button = styled.button`
  border: none;
  color: #fff;
  background: #1d96ff;
  padding: 5px 15px;
  border-radius: 3px;
`

export const ButtonReset = styled.button`
  ${Button} {
  }
`

export const ButtonSubmit = styled.button`
  ${Button} {
  }
`
