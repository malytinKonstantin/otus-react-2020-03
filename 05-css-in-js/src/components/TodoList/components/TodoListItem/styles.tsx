import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const Wrapper = styled.div`
  margin-bottom: 10px;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
`

export const Title = styled.div`
  font-family: monospace;
  font-size: 16px;
  color: #000;
  display: flex;
  cursor: pointer;
`

export const Checkbox = styled.div`
  width: 16px;
  flex-basis: 6px;
  min-width: 16px;
  height: 16px;
  border: 1px solid blue;
  border-radius: 3px;
  margin-right: 10px;
  position: relative;

  &:before {
    content: '';
    position: absolute;
    top: 3px;
    left: 3px;
    width: 10px;
    height: 10px;
    border-radius: 2px;
    background-color: ${({ isCompleted }) =>
      isCompleted ? 'blue' : 'transparent'};
  }

  &:hover {
    cursor: pointer;
  }
`

const ButtonBase = css`
  border: none;
  color: #fff;
  padding: 3px 10px;
  border-radius: 3px;

  &:active {
    outline: none;
    box-shadow: none;
  }
`

export const Button = styled.button`
  ${ButtonBase}
  background-color: #0eb902;
`

export const ButtonRemove = styled.button`
  ${ButtonBase}
  background-color: #ec2c2c;
  margin-left: 10px;
`
