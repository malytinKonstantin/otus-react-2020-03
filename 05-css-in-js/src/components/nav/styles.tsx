import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const NavWrapper = styled.nav`
  display: flex;
  margin-bottom: 30px;
`

export const NavList = styled.ul`
  display: flex;
  align-items: center;
  list-style-type: none;
  padding: 0;
  margin: 0;

  & > * {
    margin-right: 15px;
  }
`

export const User = styled.div`
  margin-left: auto;
  display: flex;
  align-items: center;
`

export const UserName = styled.div`
  margin-right: 15px;
`
