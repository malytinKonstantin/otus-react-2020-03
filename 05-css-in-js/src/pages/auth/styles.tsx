import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const Row = styled.div`
  margin-bottom: 15px;
`

export const Wrapper = styled.div`
  margin: 0 auto;
  padding-top: 100px;
  width: 200px;
`

export const Label = styled.label`
  display: block;
`

export const Input = styled.input`
  width: 100%;
`
