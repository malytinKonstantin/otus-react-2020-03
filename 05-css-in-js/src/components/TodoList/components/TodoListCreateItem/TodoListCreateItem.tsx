import React, { useState, useCallback } from 'react'
import { Formik, Form, Field } from 'formik'
import { Wrapper, TextField, ButtonReset, ButtonSubmit } from './styles'

import type { TodoListItem } from '../TodoListItem'

export interface TodoListCreateItemProps {
  /*
   * callback from parent for submit created item
   */
  onCreate: (listItem: TodoListItem) => void
}

export const TodoListCreateItem: React.FC<TodoListCreateItemProps> = (
  props,
) => {
  const handleSubmit = (values, form) => {
    const item = {
      id: Math.random(),
      title: values.title,
      isCompleted: false,
    }
    props.onCreate(item)
    form.resetForm()
  }

  return (
    <Formik
      onSubmit={handleSubmit}
      initialValues={{
        title: '',
      }}
    >
      {(props) => (
        <Form>
          <Wrapper>
            <Field
              type="text"
              name="title"
              id="title"
              component={TextField}
              value={props.values.title}
              onChange={props.handleChange}
            />
            <ButtonSubmit type="submit">добавить</ButtonSubmit>
            <ButtonReset type="button" onClick={props.resetForm}>
              сброс
            </ButtonReset>
          </Wrapper>
        </Form>
      )}
    </Formik>
  )
}
