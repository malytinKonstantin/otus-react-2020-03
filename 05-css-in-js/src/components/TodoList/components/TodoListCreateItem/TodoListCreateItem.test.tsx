import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { TodoListCreateItem } from './TodoListCreateItem'
import { TextField } from './styles'

describe('<TodoListCreateItem />', () => {
  let nextItem = null
  const title = 'title test item'

  const wrapper = shallow(
    <TodoListCreateItem
      onCreate={jest.fn((item) => {
        nextItem = item
      })}
    />,
  )

  it('renders snapshot', () => {
    expect(
      renderer.create(<TodoListCreateItem onCreate={jest.fn} />).toJSON(),
    ).toMatchSnapshot()
  })

  it('Default render', () => {
    expect(wrapper.find(TextField)).toHaveLength(1)
    expect(wrapper.find('.btn-reset')).toHaveLength(1)
    expect(wrapper.find('.btn-add')).toHaveLength(1)
  })

  it('input change event', () => {
    wrapper.find(TextField).simulate('change', { target: { value: title } })
    expect(wrapper.find(TextField).prop('value')).toBe(title)
  })

  it('reset button click', () => {
    wrapper.find('.btn-reset').simulate('click')
    expect(wrapper.find(TextField).prop('value')).toBe('')
  })

  it('add button click', () => {
    wrapper.find(TextField).simulate('change', { target: { value: title } })
    wrapper.find('.btn-add').simulate('click')
    expect(nextItem).not.toEqual(null)
    expect(nextItem.title).toBe(title)
    expect(nextItem.isCompleted).toBe(false)
  })
})
