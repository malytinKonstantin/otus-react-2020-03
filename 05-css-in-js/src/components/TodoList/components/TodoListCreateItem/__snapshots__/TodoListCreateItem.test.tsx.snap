// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListCreateItem /> renders snapshot 1`] = `
.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 15px;
}

.emotion-3 * {
  margin-right: 15px;
}

.emotion-0 {
  padding: 5px 10px;
  border-radius: 5px;
  box-shadow: none;
  border: 1px solid #afafaf;
}

.emotion-1 {
  border: none;
  color: #fff;
  background: #1d96ff;
  padding: 5px 15px;
  border-radius: 3px;
}

<div
  className="emotion-3"
>
  <input
    className="emotion-0"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn-add emotion-1"
    onClick={[Function]}
  >
    добавить
  </button>
  <button
    className="btn-reset emotion-1"
    onClick={[Function]}
  >
    сброс
  </button>
</div>
`;
